src/libraries/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -112,11 +122,37 @@ private async Task<bool> FillBufferAsync(Memory<byte> buffer, CancellationToken
 
             return true;
         }
+#else
+        private async Task<bool> FillBufferAsync(byte[] buffer, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            int readBytes = await _connectionStream.ReadAsync(buffer, 0, buffer.Length, cancellationToken).ConfigureAwait(false);
+            if (readBytes == 0)
+            {
+                return false;
+            }
+
+            buffer = buffer.Skip(readBytes).ToArray();
+            while (buffer.Length > 0)
+            {
+                readBytes = await _connectionStream.ReadAsync(buffer, 0, buffer.Length, cancellationToken).ConfigureAwait(false);
+                if (readBytes == 0)
+                {
+                    throw new Exception("Connection closed when expecting more data.");
+                }
+
+                buffer = buffer.Skip(readBytes).ToArray();
+            }
+
+            return true;
+        }
+#endif
 
         public async Task<Frame> ReadFrameAsync(TimeSpan timeout)
         {
-            using CancellationTokenSource timeoutCts = new CancellationTokenSource(timeout);
-            return await ReadFrameAsync(timeoutCts.Token).ConfigureAwait(false);
+            using (CancellationTokenSource timeoutCts = new CancellationTokenSource(timeout))
+            {
+                return await ReadFrameAsync(timeoutCts.Token).ConfigureAwait(false);
+            }
```

Comment:
Is the implication here that we can not use anything beyond C# 7 in our tests from here on? I'm not loving that...

Reply:
Not in all tests, only in those which must be run on Framework. I don't see a way around this, but if you have some ideas, please advice.

Reply:
Language version's been bumped to 8.0.